src/display-wayland.cc

```diff
@@ -926,8 +923,9 @@ void display_output_wayland::clear_text(int exposures) {
 
   cairo_set_source_rgba(window->cr, color.red / 255.0, color.green / 255.0,
                         color.blue / 255.0, color.alpha / 255.0);
-  cairo_set_operator(window->cr, CAIRO_OPERATOR_SOURCE);
-  cairo_paint(window->cr);
+  cairo_set_operator(window->cr, CAIRO_OPERATOR_CLEAR);
+  cairo_rectangle(window->cr, 0, 0, window->rectangle.width(),
+                  window->rectangle.height());
```

Comment:
This seems more correct - [SO answer](https://stackoverflow.com/a/53253583/3837417).

Resetting to `OVER` isn't needed because of `cairo_restore`.